src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/DisableShortsOnStartupPatch.kt

```diff
@@ -0,0 +1,79 @@
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.instruction.OneRegisterInstruction
+import org.jf.dexlib2.iface.instruction.TwoRegisterInstruction
+import org.jf.dexlib2.iface.instruction.formats.Instruction21c
+import org.jf.dexlib2.iface.reference.StringReference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("disable-startup-shorts-player")
+@Description("Disable playing YouTube Shorts when launching YouTube.")
+@StartupShortsResetCompatibility
+@Version("0.0.1")
+class DisableShortsOnStartupPatch : BytecodePatch(
+    listOf(
+        ActionOpenShortsFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
+            SwitchPreference(
+                "revanced_startup_shorts_player_enabled",
+                StringResource("revanced_startup_shorts_player_title", "Disable shorts player at app startup"),
+                false,
+                StringResource("revanced_startup_shorts_player_summary_on", "Shorts player is disabled at app startup"),
+                StringResource("revanced_startup_shorts_player_summary_off", "Shorts player is enabled at app startup")
+            )
+        )
+
+        val actionOpenShortsMethod = ActionOpenShortsFingerprint.result!!.mutableMethod
+        val actionOpenShortsIstructions = actionOpenShortsMethod.implementation!!.instructions
+
+        val constStringIndex = actionOpenShortsIstructions.indexOfFirst {
+            it.opcode == Opcode.CONST_STRING &&
+            ((it as Instruction21c).reference as StringReference).string == "com.google.android.youtube.action.open.shorts"
+        }
+        val moveResultIndex = actionOpenShortsIstructions.subList(constStringIndex, actionOpenShortsIstructions.size - 1).indexOfFirst {
+            it.opcode == Opcode.MOVE_RESULT
+        } + constStringIndex
+        val iPutBooleanIndex = actionOpenShortsIstructions.subList(moveResultIndex, actionOpenShortsIstructions.size - 1).indexOfFirst {
+            it.opcode == Opcode.IPUT_BOOLEAN
+        } + moveResultIndex
+
+        val moveResultRegister = (actionOpenShortsIstructions[moveResultIndex] as OneRegisterInstruction).registerA
+        val iPutBooleanRegister = (actionOpenShortsIstructions[iPutBooleanIndex] as TwoRegisterInstruction).registerA
+
+        actionOpenShortsMethod.addInstructions(
+            moveResultIndex + 1, """
+            invoke-static { }, Lapp/revanced/integrations/patches/DisableStartupShortsPlayerPatch;->disableStartupShortsPlayer()Z
+            move-result v1
+            if-eqz v1, :cond_startup_shorts_reset
+            if-nez v$moveResultRegister, :cond_startup_shorts_reset
+            const/4 v$iPutBooleanRegister, 0x0
+            const/4 v$moveResultRegister, 0x1
```

Comment:
It shouldn't be necessary for all this code for 3 registers. Instead use an opcode pattern as an anchor in the instructions to find the registers in a better way.